### PR TITLE
services: add account sponsorship service

### PR DESCRIPTION
**Pull request by CaioTeixeira95**
### What

Add the account sponsorship service and the `POST /tx/create-sponsored-account` endpoint.

### Why

It handles the account creation for the wallet.

### Known limitations

We should leverage Channel Accounts to have a high transaction rate.

#### PR Structure

- [x] It is not possible to break this PR down into smaller PRs.
- [x] This PR does not mix refactoring changes with feature changes.
- [x] This PR's title starts with name of package that is most changed in the PR, or `all` if the changes are broad or impact many packages.

#### Thoroughness

- [x] This PR adds tests for the new functionality or fixes.
- [x] All updated queries have been tested (refer to [this](https://stackoverflow.com/a/29163465) check if the data set returned by the updated query is expected to be same as the original one).

#### Release

- [x] This is not a breaking change.
- [x] This is ready to be tested in development.
- [ ] The new functionality is gated with a feature flag if this is not ready for production.

### CODE REVIEW

File: internal/serve/serve.go [hunk 1]
```diff
@@ -83,7 +104,7 @@ func handler(deps handlerDeps) http.Handler {
 
 	// Authenticated routes
 	mux.Group(func(r chi.Router) {
-		r.Use(middleware.SignatureMiddleware(deps.SignatureVerifier))
+		// r.Use(middleware.SignatureMiddleware(deps.SignatureVerifier))
```
[the remark this answers is not in the document]
Comment (CaioTeixeira95) [reply internal/serve/serve.go:107]: Thank you for this! It shouldn't be commented out.

File: cmd/serve.go [hunk 2]
```diff
@@ -43,11 +53,80 @@ func (c *serveCmd) Command() *cobra.Command {
...
+				return fmt.Errorf("setting values of config options: %w", err)
+			}
+
+			signatureClient, err := signing.NewEnvSignatureClient(distributionAccountPrivateKey, networkPassphrase)
+			if err != nil {
+				log.Fatalf("Error instantiating Env signature client: %s", err.Error())
+			}
+
+			cfg.SignatureClient = signatureClient
+			cfg.HorizonClient = &horizonclient.Client{
```
Comment (daniel-burghardt) [inline cmd/serve.go:123]: I understand that we want to instantiate the signature client here to minimize the exposure of private keys throughout the application, but do you think that we should instantiate other clients here too as opposed to in the `getHandlerDeps` function? Curious to your thoughts.

**Review by daniel-burghardt — APPROVED** [review]
This is beautiful! Thanks a lot Caio! 🚀 
Left some minor comments and questions to have a look before merging, but great job!! ✅

File: internal/services/account_sponsorship_service.go [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+	SignatureClient       signing.SignatureClient
+	HorizonClient         horizonclient.ClientInterface
+	MaxSponsoredThreshold int
+	BaseFee               int64
+}
+
+var _ AccountSponsorshipService = (*accountSponsorshipService)(nil)
+
+func (s *accountSponsorshipService) SponsorAccountCreationTransaction(ctx context.Context, accountToSponsor string, signers []entities.Signer, supportedAssets []entities.Asset) (string, string, error) {
+	// check the accountToSponsor does not exist on Stellar
```
[the remark this answers is not in the document]
Comment (CaioTeixeira95) [reply internal/services/account_sponsorship_service.go:39]: The official [docs](https://tip.golang.org/doc/comment) have examples of both cases:

```
package io

// A LimitedReader reads from R but limits the amount of
// data returned to just N bytes. Each call to Read
// updates N to reflect the new amount remaining.
// Read returns EOF when N <= 0.
type LimitedReader struct {
    R   Reader // underlying reader
    N   int64  // max bytes remaining
}
```

But I'm ok with having a pattern. I'll do that.

File: internal/services/account_sponsorship_service.go [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+)
+
+var (
+	ErrAccountAlreadyExists   = errors.New("account already exists")
+	ErrSponsorshipLimitExceed = errors.New("sponsorship limit exceed")
+)
+
+const (
+	// Sufficient to cover three average ledger close time.
+	CreateAccountTxnTimeBounds             = 18
```
[the remark this answers is not in the document]
Comment (CaioTeixeira95) [reply internal/services/account_sponsorship_service.go:21]: Yes.

File: cmd/serve.go (hunk 2 above)
Comment (CaioTeixeira95) [reply cmd/serve.go:123]: We can instantiate on `getHandlerDeps`.

File: cmd/serve.go [hunk 5]
```diff
@@ -43,11 +51,67 @@ func (c *serveCmd) Command() *cobra.Command {
...
+		{
+			Name:           "distribution-account-private-key",
+			Usage:          "The Distribution Account private key.",
+			OptType:        types.String,
+			CustomSetValue: utils.SetConfigOptionStellarPrivateKey,
+			ConfigKey:      &distributionAccountPrivateKey,
+			Required:       true,
+		},
+		{
+			Name:        "max-sponsored-base-reservers",
```
Comment (daniel-burghardt) [inline cmd/serve.go:72]:
```suggestion
			Name:        "max-sponsored-base-reserves",
```

**Review by daniel-burghardt — APPROVED** [review]

**Pull request merged**